src/FormatNumber/Parser.elm

```diff
@@ -378,3 +380,61 @@ parse locale original =
                 | prefix = locale.zeroPrefix
                 , suffix = locale.zeroSuffix
             }
+
+
+{-| Given a `Locale` parses a `String` into a `Maybe Float`
+-}
+parseString : Locale -> String -> Maybe Float
+parseString locale value =
+    let
+        isNegative : Bool
+        isNegative =
+            negativePrefixMatch && negativeSuffixMatch
+
+        negativePrefixMatch : Bool
+        negativePrefixMatch =
+            value
+                |> String.startsWith locale.negativePrefix
+
+        negativeSuffixMatch : Bool
+        negativeSuffixMatch =
+            value
+                |> String.endsWith locale.negativeSuffix
+
+        chrs : Regex.Regex
+        chrs =
+            "[^\\d]"
+                |> Regex.fromString
+                |> Maybe.withDefault Regex.never
+
+        splitValue : String -> List String
+        splitValue nbr =
+            nbr
+                |> String.split locale.decimalSeparator
+                |> List.map (Regex.replace chrs (\_ -> ""))
```

Comment:
`chrs` and `nbrs` are not readable and in Elm we usually try to use meaningful variable names. Also, I don't think it character or number describes what they are. I think we can wrap this logic in more meaningful abstractions and names (this is just pseudo-code as an example, I haven't run this code):

```suggestion
         notDigits : Regex.Regex
         notDigits =
             "\\D"
                 |> Regex.fromString
                 |> Maybe.withDefault Regex.never

        onlyDigits : String -> String
        onlyDigits =
            Regex.replace notDigits (\_ -> "")
            
        splitDecimals : String -> List String
        splitDecimals value =
           value
                |> String.split locale.decimalSeparator
                |> List.map onlyDigits
```

Reply:
I am coming from Go coding background. It is notional to provide small names such as r, s, and i. Corrected the code as per advise. Thanks for feedback.